analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -113,20 +112,20 @@ public sealed class SetLocaleForDataTypes : SonarDiagnosticAnalyzer
                 ? semanticModel.GetSymbolInfo(objectCreation.Type).Symbol
                 : semanticModel.GetSymbolInfo(constructorCall).Symbol?.ContainingType;
 
-        private static SyntaxNode GetAssignmentTargetVariable(SyntaxNode objectCreation) =>
+        private static ImmutableArray<SyntaxNode> GetAssignmentTargetVariables(SyntaxNode objectCreation) =>
             objectCreation.GetFirstNonParenthesizedParent() switch
             {
-                AssignmentExpressionSyntax assignment => assignment.Left,
-                EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration => variableDeclaration.Variables.Last(),
-                _ => null
+                AssignmentExpressionSyntax assignment => assignment.AssignmentTargets(),
```

Comment:
The direct parent of the `new` is an assignment. In this case `assignment.Left` is the right choice